packages/directlinespeech/src/playCognitiveServicesStream.js

```diff
@@ -56,6 +58,49 @@ function abortToReject(signal) {
   });
 }
 
+// In a 2 channel audio (A/B), the data come as interleaved like "ABABABABAB".
+// This function will take "ABABABABAB" and return an array ["AAAAA", "BBBBB"].
+function deinterleave(channelInterleavedAudioData, { channels }) {
+  const multiChannelArrayBuffer = new Array(channels);
+  const frameSize = channelInterleavedAudioData.length / channels;
+
+  for (let channel = 0; channel < channels; channel++) {
+    const audioData = new Float32Array(frameSize);
+
+    multiChannelArrayBuffer[channel] = audioData;
+
+    for (let offset = 0; offset < frameSize; offset++) {
+      audioData[offset] = channelInterleavedAudioData[offset * channels + channel];
+    }
+  }
+
```

Comment:
delete extra space

Reply:
Can't see where is the extra space. 👀

Reply:
lol, I swear there was an extra line (not space) there yesterday. sorry about that. Resolving.